### ISSUE TITLE
Fix S2368 FP: Do not raise on extension methods when "this" is multidimensional array

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#7329]
### Description

More details can be found on [this community ticket](https://community.sonarsource.com/t/about-csharpsquid-s2368-and-extensions-methods/91769)

### Repro steps

```csharp
public static void MatrixForEach<T>(this T[,] dataMatrix, Action<T, int, int> action) // Do not raise here, this extension method is specifically made for multi-d matrices
{ 
  // code
}
```

### Expected behavior

Do not raise.

### Actual behavior

Raises.

**Comment by gregory-paidis-sonarsource** [comment]
Not sure if this should be marked as `FP` or `Improvement`, but the description of a FP is "rule is raising when it should NOT", so I think this should be taken care of as part of a hardening sprint (as we usually do we FP/FNs)
